dojo/tools/sonatype/parser.py

```diff
@@ -1,154 +1,95 @@
-            "threatCategory", "CVSS vector not provided. "
-        ).title()
-        status = main_finding["status"]
-        main_finding.get("severity", "No CVSS score yet.")
-        if "pathnames" in vulnerability:
-            file_path = " ".join(vulnerability["pathnames"])[:1000]
-        else:
-            file_path = ""
-
-        # create the finding object
-        finding = Finding(
-            title=finding_title,
-            test=test,
-            severity=severity,
-            description=finding_description,
-            mitigation=status,
-            references="{}\n{}\n".format(
-                main_finding["url"], "\n".join(references)
-            ),
-            false_p=False,
-            duplicate=False,
-            out_of_scope=False,
-            mitigated=None,
-            file_path=file_path,
-            impact=threat_category,
-            static_finding=True
-        )
-        if vulnerability_id:
-            finding.unsaved_vulnerability_ids = [vulnerability_id]
-        finding.description = finding.description.strip()
-
-        return finding
+
+                for security_issue in component["securityData"]["securityIssues"]:
+                    finding = get_finding(security_issue, component, test)
+                    findings.append(finding)
+
+        return findings
+
+
+def get_finding(security_issue, component, test):
+
+    severity = get_severity(security_issue)
+    threat_category = security_issue.get("threatCategory", "CVSS vector not provided. ").title()
+    status = security_issue["status"]
+    reference = security_issue["url"]
+
+    identifier = ComponentIdentifier(component)
+    title = f"{security_issue['reference']} - {identifier.component_id}"
+
+    finding_description = f"Hash {component['hash']}\n\n"
+    finding_description += identifier.component_id
+    finding_description = finding_description.strip()
+
+    finding = Finding(
+        test=test,
+        title=title,
+        description=finding_description,
+        component_name=identifier.component_name,
+        component_version=identifier.component_version,
+        severity=severity,
+        mitigation=status,
+        references=reference,
+        impact=threat_category,
+        static_finding=True
+    )
+    if "cwe" in security_issue:
+        finding.cwe = security_issue["cwe"]
+
+    if "cvssVector" in security_issue:
+        finding.cvssv3 = security_issue["cvssVector"]
+
+    if "pathnames" in component:
+        finding.file_path = " ".join(component["pathnames"])[:1000]
+
+    if security_issue.get("source") == "cve":
+        vulnerability_id = security_issue.get("reference")
+        finding.unsaved_vulnerability_ids = [vulnerability_id]
+
+    if status == "Open":
+        pass
+    elif status == "Acknowledged":
+        finding.verified = True
+    elif status == "Not Applicable":
+        finding.verified = True
+        finding.out_of_scope = True
```

Comment:
I'm not too familiar with Sonatype, but couldn't this also arguably be a `false_p` based on how the user manages things in Sonatype? We're starting to lean away from automatically marking Findings `out_of_scope` / `false_p` / `risk_accepted` at the parser level if it's ambiguous, since it can vary based on user assumptions. For example, [with a similar case in the BugCrowd parser](https://github.com/DefectDojo/django-DefectDojo/pull/9517/files#diff-78adb39511a2d994c7c200bb18f4e89fd0ca0769f94e41410edbcfe156803e30R148), we opted to mark those findings as inactive/"Info" severity.

Reply:
Yes you are right. I was unsure about that too. This is more our business logic and would probably be better off in our own code than in the official DefectDojo repo. I remove this block and the associated test.